stdlib/InteractiveUtils/src/macros.jl

```diff
@@ -21,25 +22,33 @@ function rewrap_where(ex::Expr, where_params::Union{Nothing, Vector{Any}})
     Expr(:where, ex, esc.(where_params)...)
 end
 
-function get_typeof(@nospecialize ex)
-    # Always unescape to get the core expression, then reescape and esc
-    original_ex = ex
-    ex = Meta.unescape(ex)
+function reescape(f::Function, @nospecialize ex)
+    isa(ex, Expr) || return f(ex)
+    unescaped = Meta.unescape(ex)
+    new = f(unescaped)
+    return reescape(new, ex)
+end
```

Comment:
Why are you vendoring a buggy copy of Meta.reescape now?

Reply:
What is buggy with its implementation? My understanding is that the pattern is:
- Get the escaped argument (with `Meta.unescape`)
- Process it to possibly generate a new `Expr` (that's "user"-defined)
- Re-apply the original hygiene environment (escapes, hygienic scopes etc)

Reply:
The `reescape(f::Function, ex)` method is specifically to apply the pattern from #59239 where we [unescape to get the core expression, then reescape](https://github.com/JuliaLang/julia/blob/8190fe87b5dffd7dd1f15939aa7c14e5f7332be8/stdlib/InteractiveUtils/src/macros.jl#L107). I'd be happy to change it if it does something wrong, but the pattern it automates looks correct to me.

I agree that the other (copied) `reescape` method can be removed. I have a question though, why do we remove `:var"hygienic-scope"` with `unescape` but [don't reapply it](https://github.com/JuliaLang/julia/blob/cbea8cf28e0f617333130add1c4ab2a51f6c979c/base/meta.jl#L640) with `reescape` (and apply `escape` instead regardless if the original was an `:escape` or `:var"hygienic-scope"` expression)?